acpreprocessing/neuroglancer/segmentation.py

```diff
@@ -0,0 +1,169 @@
+    """Create a neuroglancer precomputed segmentation volume from a tiff stack.
+    
+       source_path: directory underwhich `swc_files_nm` will be found
+                    (Example path: /data/olga/MN7_RH_3_2_S35_220127_high_res/Pos10_10)
+       out_path: directory where new the new cloud volume segmentation layer should be generated
+                    (Example path: /data/out/MN7_RH_3_2_S35_220127_high_res/Pos10_10)
+       chunk_size: list or array with chunk size for segmentation layer
+                    (Example: (128,128,128))
+       resolution: voxel resolution, in nm (passed on to neuroglancer via 'info' file)
+                    (Example: (4,4,40))
+    """
+
+    # Read tiffstack into memory
+    files = glob.glob(f"{source_path}/Segmentation_labeled/*.tif")
+    files = sorted(files)
+    data = None
+    for layer in range(len(files)):
+        image = tifffile.imread(files[layer])
+        # Allocate array if it has not been yet, using the number of files and dimensions of first file
+        if data is None:
+            data = np.zeros(shape=(image.shape[1], image.shape[0], len(files)), dtype=image.dtype)
+        data[:, :, layer] = image.T   # Swap X & Y to match the SWC files? (not sure why this is needed)
+ 
+    info = CloudVolume.create_new_info(
+        num_channels    = 1,
+        layer_type      = 'segmentation',
+        data_type       = 'uint64', # Channel images might be 'uint8'
+        # raw, png, jpeg, compressed_segmentation, fpzip, kempressed, compresso
+        encoding        = 'compressed_segmentation', 
+        resolution      = resolution, # Voxel scaling, units are in nanometers
+        voxel_offset    = [0, 0, 0], # x,y,z offset in voxels from the origin
+        # Pick a convenient size for your underlying chunk representation
+        # Powers of two are recommended, doesn't need to cover image exactly
+        chunk_size      = chunk_size, # units are voxels
+        volume_size     = data.shape, # e.g. a cubic millimeter dataset
+        skeletons       = 'skeletons'
+        )
+
+    vol = CloudVolume(f'file://{out_path}', info=info, compress='', cache=False)
+    print("Creating cloud volume: ", vol.info)
+    vol.commit_info()
+    vol.commit_provenance()
+    vol[:,:,:] = data.astype(np.uint64)
+
+
+def generate_ngl_skeletons(source_path, out_path):
+    """Generate skeletons from SWC files.
+       This currently assumes the neuroglancer precomputed volume has already been generated by generate_ngl_segmentation.
+
+       source_path: directory underwhich `swc_files_nm` will be found.
+       out_path: directory with the previously generated segmentation layer.
+       
+    """
+    # There a few cloud-volume bugs we are working around:
+    #   * The info file is not generated
+
+    vol = CloudVolume(f'file://{out_path}', compress='')
+    files = glob.glob(f"{source_path}/swc_files_nm/*.swc")
+    files = sorted(files)
+    skel_dir = os.path.join(out_path, "skeletons")
```

Comment:
Is there a complication in using `vol.skeleton.upload` or `vol.skeleton.upload_raw`?  If not, this should make e.g. sharded support easier too.

Reply:
I will address this after submitting a small PR to cloud-volume to allow uncompressed skeleton generation.